rule-engine/rule-engine-components/src/main/java/org/thingsboard/rule/engine/profile/AlarmRuleState.java

```diff
@@ -138,10 +139,32 @@ private boolean isActive(long eventTs) {
         switch (alarmRule.getSchedule().getType()) {
             case ANY_TIME:
                 return true;
-            case SPECIFIC_TIME:
-                return isActiveSpecific((SpecificTimeSchedule) alarmRule.getSchedule(), eventTs);
-            case CUSTOM:
-                return isActiveCustom((CustomTimeSchedule) alarmRule.getSchedule(), eventTs);
+            case SPECIFIC_TIME: {
+                SpecificTimeSchedule defaultSchedule = (SpecificTimeSchedule) alarmRule.getSchedule();
+                EntityKeyValue dynamicValue = getDynamicValue(data, defaultSchedule.getDynamicValue());
+
+                SpecificTimeSchedule schedule;
+                try {
+                    schedule = JsonConverter.parse(dynamicValue.getJsonValue(), SpecificTimeSchedule.class);
+                } catch (Exception e) {
+                    schedule = defaultSchedule;
+                }
+
+                return isActiveSpecific(schedule, eventTs);
+            }
+            case CUSTOM: {
+                CustomTimeSchedule defaultSchedule = (CustomTimeSchedule) alarmRule.getSchedule();
+                EntityKeyValue dynamicValue = getDynamicValue(data, defaultSchedule.getDynamicValue());
+
+                CustomTimeSchedule schedule;
+                try {
```

Comment:
code duplication. extract the method to parse the Schedule JSON. Also, the dynamicValue may be null. In this case you have NPE in line: `schedule = JsonConverter.parse(dynamicValue.getJsonValue(), CustomTimeSchedule.class);`.